src/Syntax/IMPEG.hs

```diff
@@ -328,7 +328,28 @@ instance (Typeable1 p, Data (p (Located (Type tyid))), Data tyid, Data typaram)
          Data (Program p tyid typaram) where
   gfoldl k z (Program x1 x2 x3) = z Program `k` x1 `k` x2 `k` x3
 
+
+
 -- Rather than manually defining these instances, we use Template Haskell to do it for us.
+
+#ifdef MIN_VERSION_GLASGOW_HASKELL && MIN_VERSION_GLASGOW_HASKELL(8,2,1,0)

```

Comment:
I am not entirely sure if this will work for GHC versions < 8.2.1.0
We need 2 conditions to be true there. One is for MIN_VERSION_GLASGOW_HASKELL to be defined, other
MIN_VERSION_GLASGOW_HASKELL to be atleast 8.2.1.0